### DISCUSSION TITLE
Please improve image browsing function in mobile phone's browser

### DISCUSSION THREAD

**Discussion by niubility000** [filebrowser/filebrowser#1200] [Ideas]
I just checked another one in github: cloudreve. Its function is very simple. It has some bugs, is not good for me. But when you use a mobile phone's browser to visit the webpage, the thumbnail view is much clearer, and to browse the fullsize images, you can swipe the screen to switch images. It's also based on mobile phone's browser, but looks pretty good. 
Can filebrowser design the webpage like that?

**Comment by niubility000** [comment]
![Screenshot_20201224-234730](https://user-images.githubusercontent.com/76441520/103097990-2bb5fe80-4644-11eb-871f-27aff47db5de.png)
We can swipe to show next pic, or double click to enlarge.

**Reply by WeidiDeng** [reply]
I'll consider it when I have time.

**Comment by WeidiDeng** [comment]
Fixed in [this](https://github.com/filebrowser/filebrowser/pull/1224).

**Reply by niubility000** [reply]
Thank you soooo much!  I will wait for the next version of filebrowser. It will be a perfect one for me.

**Reply by niubility000** [reply]
> Fixed in [this](https://github.com/filebrowser/filebrowser/pull/1224).

看了好多网页网盘的设计，图片浏览功能都只能局限在浏览器里。cloudreve，可道云用的是同一种方式，还不错。不过最好把上下那两个黑条弄成完全透明的（越透明越好），看起来图片就完整了，会专业很多。右上角的X是白色的，底下是黑底，应该也能看清楚。

**Reply by WeidiDeng** [reply]
我写的并没有下面的黑条，上面的黑条只有移动上去才有显示。

**Reply by niubility000** [reply]
> 我写的并没有下面的黑条，上面的黑条只有移动上去才有显示。

😀

**Reply by niubility000** [reply]
> 我写的并没有下面的黑条，上面的黑条只有移动上去才有显示。

哇，谢谢谢谢！另外，最好上面的黑条上再加个 旋转 按钮，有些图片显示是横着的或倒着的。可以每按一次旋转90度。再加个按钮能显示图片metadata信息（就像filerun那个一样）就更牛叉了。😀

**Reply by WeidiDeng** [reply]
Meta信息，能不能截图举例，除了分辨率以外的不太清楚

**Reply by niubility000** [reply]
> Meta信息，能不能截图举例，除了分辨率以外的不太清楚

这是filerun的展示网站：https://demo.filerun.co/?username=admin&password=admin
，里面随便点个图片，点上面的叹号，就出来信息了，拍摄时间 date taken最有用。（其实没这个功能也没啥。)

另外，有个小bug有些影响使用。上传文件和文件夹怎么操作一样啊？选上传文件夹后并不能选中文件夹，只能选里面的文件，这样要想上传一个1000多张照片的文件夹就几乎不可能了。

**Reply by WeidiDeng** [reply]
我用Chrome是可以上传的文件夹的，那部分不是我写的，不太清楚。个人其实更偏向于用webdav进行上传（客户端全面），不过目前不支持webdav方式访问。

拍摄时间这个需要在后端进行处理，这要慢慢来。

**Reply by niubility000** [reply]
我在chrome下也不能选文件夹，这个是调用的第三方app，我这个选不了。大量上传用ftp，samba就行。

**Reply by WeidiDeng** [reply]
手机端上传的？我刚才试了试，会进入文件浏览器，然后就不能选择文件了。

**Reply by niubility000** [reply]
安卓qq浏览器可以，但只能选中文件。

还有个bug，找了半天才想起来。开启单击模式后，复制和移动时仍然是双击，在手机上很难操作。

**Reply by WeidiDeng** [reply]
手机浏览器的兼容性以后再说。那个单击bug我先改了

**Reply by niubility000** [reply]
> 手机浏览器的兼容性以后再说。那个单击bug我先改了

谢谢😊。手机浏览器上传文件夹估计很难做到，我去瞅瞅别的web网盘有能做到的吗。

**Reply by niubility000** [reply]
> Fixed in [this](https://github.com/filebrowser/filebrowser/pull/1224).

老大，这个功能似乎没有merge到master里，可能是因为改动太大了？
看了几个其他项目的设计，有个蓝眼云盘，演示版：https://tank.eyeblue.cn/user/login，用户名demo，密码123456。这个在手机浏览器里浏览图片时功能就很不错。能够在网页里尽量占满空间，背景是黑色不透明的，看起来很清晰。可以滑动切换图片。双击后直接放大图片，一直往右滑会切换到下一张图片。就是上面的黑边有些不美观。可以参考nextcloud的设计，上下都没有黑条，是全透明的，按钮是白色的，挺清晰。演示版在：try.nextcloud.com，密码是demo。这两个网站在github也都是开源的。
filebrowser最大的好处是安装极其方便，浏览速度快设计结构清晰。而且能浏览硬盘里已有的文件，不打乱原有目录结构。能把手机浏览器里的图片浏览功能再加强些就好了。

**Reply by niubility000** [reply]
> Fixed in [this](https://github.com/filebrowser/filebrowser/pull/1224).

老大，我把你修改的那五个文件替换上了，自己用windows编译了一下（学了一天才搞明白），用命令运行了一下。
.\filebrowser -d filebrowser.db --disable-preview-resize --cache-dir cache
在电脑浏览器里很完美。在手机浏览器里有一些小不足：
1. 滑动不是很灵敏，貌似滑动得足够长，而且一定要水平滑动才行。滑动时距离短了，或略向下歪了，只是晃动一下图片，但不切换到下一张。其实在这种预览状态下，向左划，不论多大，不论什么方向都可以直接切换到下一张了。向右划，不论多大，不论什么角度都可以直接切换到上一张了。双击的话，直接放大到原始图片。现在有时要滑动好几下才切换到下一张。能调成那个 蓝眼云盘 那样灵敏的就完美了。双击后显示原始图片时可以滑动来移动看图片细节这个挺不错的，就是向左移动图片时，移过图片边缘后，会弹到图片的最右侧，应该是弹到图片的最左侧吧！向右移动图片时效果就不错。

2. 在手机浏览器里刚开始时那个最上方的黑条是隐藏的，旋转按钮也是隐藏的，没显示，但能用，点击那个位置可以旋转，然后就能显示了。能不能让它一直显示？而且，最好把这个黑条做成nextcloud那样的透明的，配上白色的按钮也挺清晰。这样就彻底不影响看图片了。（又看了一下，电脑浏览器里这个黑条要把鼠标放上去才有显示，移开就自动隐藏。但有时候不会隐藏，反而变成透明的了，完全不遮挡图片，很漂亮，这种bug状态就特好。没必要自动隐藏。手机浏览器也弄成这个效果就好极了！！！底下的图片名称其实可以去掉，那个黑条就彻底不用了。可以用info按钮来查看图片名称。）

3. 另外，最好把这个黑条(最好是透明的)移到底下来，让旋转按钮在左下角偏中间。手机个头大，用左手单手操作，右上角的旋转按钮不好点到。
4. 浏览图片时，按左上角的x后，回到原始的文件列表页，这时没有办法自动定位到刚才浏览的图片的文件位置吗？如果能就更好了。。。

谢谢！！已经做得很漂亮了！
